patches/openwrt/0048-fix-UBNT-XM-model-detection.patch

```diff
@@ -0,0 +1,73 @@
+From: Neal Oakey <neal.oakey@bingo-ev.de>
+Date: Thu, 28 Jan 2016 20:04:00 +0100
+Subject: fix UBNT XM model detection
+
+Signed-off-by: Neal Oakey <neal.oakey@bingo-ev.de>
+
+diff --git a/target/linux/ar71xx/base-files/lib/ar71xx.sh b/target/linux/ar71xx/base-files/lib/ar71xx.sh
+index d0abf42..6a69a6e 100755
+--- a/target/linux/ar71xx/base-files/lib/ar71xx.sh
++++ b/target/linux/ar71xx/base-files/lib/ar71xx.sh
+@@ -64,6 +64,38 @@ wndr3700_board_detect() {
+ 	AR71XX_MODEL="$machine"
+ }
+ 
++ubnt_get_mtd_part_magic() {
++	ar71xx_get_mtd_offset_size_format EEPROM 4118 2 %02x
++}
++
++ubnt_xm_board_detect() {
++	local model
++	local magic
++
++	magic="$(ubnt_get_mtd_part_magic)"
++	case ${magic:0:3} in
++		"e80"|\
++		"e01")
++			model="Ubiquiti NanoStation M"
++			;;
++		"e0a")
++			model="Ubiquiti NanoStation loco M"
++			;;
++		"e1b") # Note: the M5 has not been tested!
++			model="Ubiquiti Rocket M"
++			;;
++		"e20"|\
++		"e2d")
++			model="Ubiquiti Bullet M"
++			;;
++		"e30") # Note: the M5 has not been tested!
++			model="Ubiquiti PicoStation M"
++			;;
++	esac
++
++	[ ! -z "$model" ] && AR71XX_MODEL="${model}${magic:3:1}"
++}
++
+ cybertan_get_hw_magic() {
+ 	local part
+ 
+@@ -311,6 +343,8 @@ gl_inet_board_detect() {
+ 	esac
+ }
+ 
++
++
+ ar71xx_board_detect() {
+ 	local machine
+ 	local name
+@@ -472,12 +506,14 @@ ar71xx_board_detect() {
+ 		;;
+ 	*"Bullet M")
+ 		name="bullet-m"
++		ubnt_xm_board_detect
+ 		;;
+ 	*"Loco M XW")
+ 		name="loco-m-xw"
+ 		;;
+ 	*"Nanostation M")
+ 		name="nanostation-m"
++		ubnt_xm_board_detect
```

Comment:
Please also add the new detection for the Rocket M, we'd like to switch to the proper Rocket images.